src/index.ts

```diff
@@ -63,6 +75,12 @@ interface BasePushable<T> {
    */
   push: (value: T) => this
 
+  /**
+   * Returns a promise that resolves when the underlying queue becomes empty (e.g.
+   * this.readableLength === 0).
+   */
+  onEmpty: (signal?: AbortSignal) => Promise<void>
```

Comment:
I'd be tempted to pass this in an options object but not strong feels.
```suggestion
  onEmpty: ({ signal?: AbortSignal } = {}) => Promise<void>
```

Reply:
Making the suggested change results in the lovely error:

> Initializer provides no value for this binding element and the binding element has no default value.

I've broken it out into an interface instead.